test/src/main/java/org/apache/accumulo/test/functional/FateInterleavingIT.java

```diff
@@ -0,0 +1,265 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FateInterleavingIT extends SharedMiniClusterBase {
+
+  public static class FirstOp extends ManagerRepo {
+
+    private static final long serialVersionUID = 1L;
+
+    protected static void insertTrackingData(long tid, Manager manager, String step)
+        throws TableNotFoundException, MutationsRejectedException {
+      try (BatchWriter bw = manager.getContext().createBatchWriter(FATE_TRACKING_TABLE)) {
+        Mutation mut = new Mutation(Long.toString(System.currentTimeMillis()));
+        mut.put("TXID:" + Long.toString(tid), "", step);
+        bw.addMutation(mut);
+      }
+    }
+
+    @Override
+    public long isReady(long tid, Manager manager) throws Exception {
+      Thread.sleep(500);
+      insertTrackingData(tid, manager, this.getName() + "::isReady");
+      return 0;
+    }
+
+    @Override
+    public Repo<Manager> call(long tid, Manager manager) throws Exception {
+      Thread.sleep(500);
+      insertTrackingData(tid, manager, this.getName() + "::call");
+      return new SecondOp();
+    }
+
+  }
+
+  public static class SecondOp extends FirstOp {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public Repo<Manager> call(long tid, Manager environment) throws Exception {
+      super.call(tid, environment);
+      return new LastOp();
+    }
+
+  }
+
+  public static class LastOp extends FirstOp {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public Repo<Manager> call(long tid, Manager environment) throws Exception {
+      super.call(tid, environment);
+      return null;
+    }
+  }
+
+  public static class FateInterleavingConfig implements MiniClusterConfigurationCallback {
+
+    @Override
+    public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration coreSite) {
+      cfg.setProperty(Property.MANAGER_FATE_THREADPOOL_SIZE, "1");
+    }
+  }
+
+  private static final String FATE_TRACKING_TABLE = "fate_tracking";
+
+  @BeforeAll
+  public static void setup() throws Exception {
+    SharedMiniClusterBase.startMiniClusterWithConfig(new FateInterleavingConfig());
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(FATE_TRACKING_TABLE);
+    }
+  }
+
+  @AfterAll
+  public static void teardown() throws Exception {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  @BeforeEach
+  public void before() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().deleteRows(FATE_TRACKING_TABLE, null, null);
+    }
+  }
+
+  private void waitFor(ZooStore<Manager> store, long txid) throws Exception {
+    TStatus status;
+    do {
+      store.reserve(txid);
+      try {
+        status = store.getStatus(txid);
+      } finally {
+        store.unreserve(txid, Duration.ZERO);
+      }
```

Comment:
is reserving required in order to get the status in 3.x?

Reply:
Yes.

https://github.com/apache/accumulo/blob/d0fb7f5fd215c9f60dabb123169ef2df71fc6506/core/src/main/java/org/apache/accumulo/core/fate/ZooStore.java#L403-L407